control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -0,0 +1,236 @@
+	br := r.createReconcilerForBrokerInstance(broker)
+	result := br.FinalizeKind(ctx, broker)
+
+	r.IPsLister.Unregister(types.NamespacedName{Namespace: broker.Namespace, Name: broker.Name})
+
+	return result
+}
+
+func (r *NamespacedReconciler) createReconcilerForBrokerInstance(broker *eventing.Broker) *Reconciler {
+
+	return &Reconciler{
+		Reconciler: &base.Reconciler{
+			KubeClient:             r.Reconciler.KubeClient,
+			PodLister:              r.Reconciler.PodLister,
+			SecretLister:           r.Reconciler.SecretLister,
+			SecretTracker:          r.Reconciler.SecretTracker,
+			ConfigMapTracker:       r.Reconciler.ConfigMapTracker,
+			DataPlaneConfigMapName: r.Reconciler.DataPlaneConfigMapName,
+			DataPlaneConfigFormat:  r.Reconciler.DataPlaneConfigFormat,
+			DispatcherLabel:        r.Reconciler.DispatcherLabel,
+			ReceiverLabel:          r.Reconciler.ReceiverLabel,
+
+			DataPlaneNamespace:          broker.Namespace,
+			DataPlaneConfigMapNamespace: broker.Namespace,
+
+			DataPlaneConfigMapTransformer: func(cm *corev1.ConfigMap) {
+				kafka.NamespacedDataplaneLabelConfigmapOption(cm)
+				appendBrokerAsOwnerRef(broker)(cm)
+			},
+		},
+		Env:                        r.Env,
+		Resolver:                   r.Resolver,
+		ConfigMapLister:            r.ConfigMapLister,
+		NewKafkaClusterAdminClient: r.NewKafkaClusterAdminClient,
+		BootstrapServers:           r.BootstrapServers,
+		Prober:                     r.Prober,
+	}
+}
+
+// getManifest returns the manifest that is transformed from the BaseDataPlaneManifest with the changes
+// for the given broker
+func (r *NamespacedReconciler) getManifest(broker *eventing.Broker) (mf.Manifest, error) {
+	manifest := r.BaseDataPlaneManifest
+
+	manifest, err := manifest.Transform(mf.InjectNamespace(broker.Namespace))
+	if err != nil {
+		return mf.Manifest{}, fmt.Errorf("unable to transform base dataplane manifest with namespace injection. namespace: %s, error: %v", broker.Namespace, err)
+	}
+
+	return manifest.Transform(appendNewOwnerRefsToPersisted(manifest.Client, broker))
+}
+
+func appendNewOwnerRefsToPersisted(client mf.Client, broker *eventing.Broker) mf.Transformer {
+	return func(resource *unstructured.Unstructured) error {
+		existingRefs, err := getPersistedOwnerRefs(client, resource)
+		if err != nil {
+			return err
+		}
+
+		refs, _ := appendOwnerRef(existingRefs, broker)
+		// We need to set this even when we don't append any ownerRef.
+		// That is because the resource is in a pristine state that's read from the YAML, and we need to set the
+		// existing owners to that in memory resource.
+		resource.SetOwnerReferences(refs)
+		return nil
+	}
+}
+
+func getPersistedOwnerRefs(mfc mf.Client, res *unstructured.Unstructured) ([]metav1.OwnerReference, error) {
+	retrieved, err := mfc.Get(res)
+	if err != nil && !errors.IsNotFound(err) {
+		// actual problem
+		return nil, err
```

Comment:
A bit of info on the error:
```suggestion
		return nil, fmt.Errorf("failed to get resource %s/%s: %w", res.GetNamespace(), res.GetName(), err)
```